### ISSUE TITLE
Prebuild 

### ISSUE THREAD

**Issue by zolnayagemc** [premake/premake-core#1722]
Dear Premake Team,

I'd like to suggest a small bug fix to the gmake module:

Problem: When running a makefile with the parallel option **-j10**, prebuild step is not guaranteed to finish before compiling the first object file. For example: prebuild step generates some c++ header files which will be used when compiling the $(TARGET).
                      
Solution: Target 'all' should not be dependent on $(TARGET) but $(TARGET) should be built separately after the prebuild and prelink dependencies have been built.

Code changes: Replace original code in modules/gmake/gmake_cpp.lua
_p('all: prebuild prelink $(TARGET)')
_p('\t@:')

with:
-- When make run with parallel option '-j10', prebuild and prelink should finish before starting $(TARGET)'
_p('all: prebuild prelink')
_p('\t@$(MAKE) --no-print-directory -f $(firstword $(MAKEFILE_LIST)) $(TARGET)')

Bye,
Andreas Zolnay

**Comment by nickclark2016** [comment]
If you use the gmake2 module, does it work?

**Comment by zolnayagemc** [comment]
HI Nick,

Yes, parallel building works with makefiles generated by gmake2.

Thanks,
Andreas

**Comment by nickclark2016** [comment]
Glad to hear it.  I'm going to close this out then, since we really don't support the gmake exporter anymore in lieu of the gmake2 exporter.  If you find missing functionality from gmake that is needed in gmake2, open up another issue and we'll take a look!
